roles/config-rh-sso/tasks/manage-roles.yml

```diff
@@ -17,14 +17,72 @@
+      uri:
+        url: "{{ rh_sso_protocol }}://{{ rh_sso_host }}:{{ rh_sso_port }}/auth/admin/realms/{{ role_data.targetRealm }}/roles"
+        method: POST
+        body: "{{ role_json }}"
+        body_format: json
+        status_code: 201, 409
+        headers:
+          Authorization: "Bearer {{ rh_sso_token.json.access_token }}"
+        validate_certs: "{{ rh_sso_ssl_cert_validation | default(omit) }}"
+      register: rh_sso_role_create
+    - name: "Update Role: {{ role_data.name }}"
+      uri:
+        url: "{{ rh_sso_protocol }}://{{ rh_sso_host }}:{{ rh_sso_port }}/auth/admin/realms/{{ role_data.targetRealm }}/roles/{{ role_data.name }}"
+        method: PUT
+        body: "{{ role_json }}"
+        body_format: json
+        status_code: 200, 204
+        headers:
+          Authorization: "Bearer {{ rh_sso_token.json.access_token }}"
+        validate_certs: "{{ rh_sso_ssl_cert_validation | default(omit) }}"
+      when:
+        - rh_sso_role_create.status == 409
+  when:
+    - role_data.clientRole is false
+
+- name: "Create Role: {{ role_data.name }} in Client: {{ role_data.clientName }} in Realm: {{ role_data.targetRealm }}"
+  block:
+    - name: "Get Clients in Realm: {{ role_data.targetRealm }}"
+      uri:
+        url: "{{ rh_sso_protocol }}://{{ rh_sso_host }}:{{ rh_sso_port }}/auth/admin/realms/{{ role_data.targetRealm }}/clients"
+        method: GET
+        headers:
+          Authorization: "Bearer {{ rh_sso_token.json.access_token }}"
+        validate_certs: "{{ rh_sso_ssl_cert_validation | default(omit) }}"
+      register: rh_sso_clients
+    - name: "Get Client ID for Client: {{ role_data.clientName }}"
+      set_fact:
+        client_id:  "{{ (rh_sso_clients.json | json_query(query))[0] }}"
+      vars:
+        query: "[?clientId=='{{ role_data.clientName }}'].id"
+    - name: "Create Client Role: {{ role_data.name }}"
+      uri:
+        url: "{{ rh_sso_protocol }}://{{ rh_sso_host }}:{{ rh_sso_port }}/auth/admin/realms/{{ role_data.targetRealm }}/clients/{{ client_id }}/roles"
+        method: POST
+        body: "{{ role_json }}"
+        body_format: json
+        status_code: 201, 409
+        headers:
+          Authorization: "Bearer {{ rh_sso_token.json.access_token }}"
+        validate_certs: "{{ rh_sso_ssl_cert_validation | default(omit) }}"
+      register: rh_sso_role_create
+    - name: "Update Client Role: {{ role_data.name }}"
+      uri:
+        url: "{{ rh_sso_protocol }}://{{ rh_sso_host }}:{{ rh_sso_port }}/auth/admin/realms/{{ role_data.targetRealm }}/clients/{{ client_id }}/roles/{{ role_data.name }}"
+        method: PUT
+        body: "{{ role_json }}"
+        body_format: json
+        status_code: 200, 204
+        headers:
+          Authorization: "Bearer {{ rh_sso_token.json.access_token }}"
+        validate_certs: "{{ rh_sso_ssl_cert_validation | default(omit) }}"
+      when:
+        - rh_sso_role_create.status == 409
+  when:
+    - role_data.clientRole is true
```

Comment:
Need to default here as well, perhaps something like:
```
- role_data.clientRole is defined
- role_data.clientRole is true
```

Reply:
Fixed.